Ds3/Helpers/Streams/StreamTranslator.cs

```diff
@@ -125,7 +126,18 @@ public override void Write(byte[] buffer, int offset, int count)
                 this._streamStore.Access(range.Context, stream =>
                 {
                     stream.Position = range.Range.Start;
-                    stream.Write(buffer, progress + offset, rangeLength);
+                    try
+                    {
+                        stream.Write(buffer, progress + offset, rangeLength);
+                    }
+                    catch (IOException e)
+                    {
+                        if (e.Message.StartsWith("There is not enough space on the disk"))
```

Comment:
ok, it looks like the other way is not better in my opinion
http://stackoverflow.com/questions/9293227/how-to-check-if-ioexception-is-not-enough-disk-space-exception-type
https://msdn.microsoft.com/en-us/library/system.exception.hresult(v=vs.110).aspx
@rpmoore let me know what do you think?